pkg/cvo/internal/operatorstatus.go

```diff
@@ -113,46 +158,36 @@ func waitForOperatorStatusToBeDone(client configclientv1.ClusterOperatorsGetter,
 		if available && !progressing && !failing {
 			return true, nil
 		}
-		return false, nil
-	})
-
-	if err != wait.ErrWaitTimeout {
-		return err
-	}
 
-	if lastCO == nil {
-		return &payload.UpdateError{
-			Reason:  "ClusterOperatorNotAvailable",
-			Message: fmt.Sprintf("Cluster operator %s has not yet reported success", co.Name),
-			Name:    co.Name,
-		}
-	}
-	co = lastCO
-
-	glog.V(3).Infof("ClusterOperator %s is not done; it is available=%v, progressing=%v, failing=%v",
-		co.Name,
-		resourcemerge.IsOperatorStatusConditionTrue(co.Status.Conditions, configv1.OperatorAvailable),
-		resourcemerge.IsOperatorStatusConditionTrue(co.Status.Conditions, configv1.OperatorProgressing),
-		resourcemerge.IsOperatorStatusConditionTrue(co.Status.Conditions, configv1.OperatorFailing),
-	)
-
-	if c := resourcemerge.FindOperatorStatusCondition(co.Status.Conditions, configv1.OperatorFailing); c != nil && c.Status == configv1.ConditionTrue {
-		if len(c.Message) > 0 {
-			return &payload.UpdateError{
+		if c := resourcemerge.FindOperatorStatusCondition(actual.Status.Conditions, configv1.OperatorFailing); c != nil && c.Status == configv1.ConditionTrue {
+			message := fmt.Sprintf("Cluster operator %s is reporting a failure", actual.Name)
+			if len(c.Message) > 0 {
+				message = fmt.Sprintf("Cluster operator %s is reporting a failure: %s", actual.Name, c.Message)
+			}
+			lastErr = &payload.UpdateError{
+				Nested:  errors.New(strings.ToLower(message)),
```

Comment:
We don't want to ToLower the nested message (from c.Message) if capitalization is valid - for instance, if the message said `check the API version` we wouldn't want it to be `check the api version`

Reply:
Ah, Yeah.. will fix it. :innocent: